torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -857,7 +876,12 @@ def _post_backward_hook(
                 # gradients are on the same device as required by the optimizer
                 # TODO: Investigate why `NO_SHARD` breaks correctness when
                 # using `non_blocking=True` here.
-                non_blocking = handle.uses_sharded_strategy
+                # TODO (rohan-varma): When CPU offload and optimizer overlap,
```

Comment:
Curious what the 'to-do' here is -- is a plan to use `non_blocking=True` still?

Reply:
mainly to check if we do nonblocking=True and then sync before use on CPU, is it faster (my guess is probably not).